Gravity/Gravity/Base/FileDto.cs

```diff
@@ -27,33 +25,43 @@ internal string GetMD5()
 
 	public class DiskFileDto : FileDto
 	{
+		public DiskFileDto() { }
+
+		public DiskFileDto(string filePath)
+			: base(filePath)
+		{
+			FilePath = filePath;
+		}
+		
 		public string FilePath { get; set; }
 
 		public ByteArrayFileDto StoreInMemory()
 		{
-			return new ByteArrayFileDto
-			{
-				ByteArray = File.ReadAllBytes(this.FilePath),
-				FileName = Path.GetFileName(this.FilePath)
-			};
+			return new ByteArrayFileDto(FilePath);
 		}
 
 		protected override Stream GetStream() => File.OpenRead(FilePath);
 	}
 
 	public class ByteArrayFileDto : FileDto
 	{
+		public ByteArrayFileDto() { }
+
+		public ByteArrayFileDto(string filePath)
```

Comment:
This can be confusing/misleading. Use `DiskFileDto(filePath).StoreInMemory()`, instead of having that point to this